src/boot/axios.js

```diff
@@ -26,20 +26,34 @@ let zapUpdateExceptions = (payload, statusCode, message) => {}
 // You can set this to false to not log all the roundtrips
 const log = false
 
+// Get the search string from the URL
 let search = window.location.search
 
+// Remove the '?' if present
 if (search[0] === '?') {
   search = search.substring(1)
 }
-let query = querystring.parse(search)
-if (window.sessionStorage.getItem('session_uuid') == null) {
+
+// Parse the query string using URLSearchParams for browser environments
+let query = new URLSearchParams(search)
+
+// Extract stsApplicationId and trim it
+let stsApplicationId = query.get('stsApplicationId')?.trim()
+
+// Get the current session UUID from sessionStorage
+let currentSessionUuid = window.sessionStorage.getItem('session_uuid')
+
+// If stsApplicationId exists, update the session_uuid
+if (stsApplicationId) {
+  // Concatenate the current session UUID with the stsApplicationId and store it back
+  window.sessionStorage.setItem(
+    'session_uuid',
+    currentSessionUuid + stsApplicationId
+  )
+} else if (currentSessionUuid == null) {
+  // If session_uuid doesn't exist, generate a new one
   window.sessionStorage.setItem('session_uuid', uuidv4())
```

Comment:
Inside Studio, this line is never reached, so the uuidv4() can never even get executed...